File: ayeon/가장큰수.cpp
// NP
/*
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool sortsort(int a, int b)
{
    if (a==b) return true;
    
	string aa=to_string(a);
	string bb=to_string(b);
	int s = min(aa.size(), bb.size());

    int i=0;
	while(i<s){
		if(aa[i]-'0' > bb[i]-'0') return true;
		else if (aa[i]-'0' == bb[i]-'0')
			i++;
		else return false;
	}

    if(s == aa.size())
    {
        if( bb[i-1]-'0' > bb[i]-'0' ) return true;
        else return false;
    }
    else{
        if(aa[i-1]-'0' > aa[i]-'0') return false;
        else return true;
    }
}


string solution(vector<int> numbers) {
    
    // TEST 11 
    // [0, 0, 0, 0, 0, 0] "0"
    
    int i=0;
    while(i<numbers.size())
    {
        if (numbers[i] == 0) i++;
        else break;
    }
    if(i==numbers.size()) return "0";
    
    
    // TEST
    string answer="";
    
    sort(numbers.begin(),numbers.end(),sortsort);

    for(int i=0; i<numbers.size(); i++)
    {
        answer+=to_string(numbers[i]);
    }

    return answer;
}
*/

File: minnkyoung/가장큰수.py
def solution(numbers): 
    numbers = list(map(str, numbers)) 
    numbers.sort(key=lambda x: x*3, reverse=True) 
    return str(int(''.join(numbers)))

#string 대소비교
#첫 문자부터 비교하여 동일한 인덱스에 더 작은 문자열이 나오면 작은 것임. 
#'121121121', '121212'의 경우 4번째 인덱스에서 '121212'가 더 크다고 인식함. 
    